lib/json/common.rb

```diff
@@ -126,51 +133,166 @@ class CircularDatastructure < NestingError; end
+  #   ruby = JSON.parse(source)
+  #   ruby.class # => Hash
+  # Use class \OpenStruct:
+  #   ruby = JSON.parse(source, {object_class: OpenStruct})
+  #   ruby # => #<OpenStruct a="foo", b=1.0, c=true, d=false, e=nil>
+  # Try class \Object:
+  #   # Raises NoMethodError (undefined method `[]=' for #<Object:>):
+  #   JSON.parse(source, {object_class: Object})
+  # Bad value:
+  #   # Raises TypeError (wrong argument type Symbol (expected Class)):
+  #   JSON.parse(source, {object_class: :foo})
+  #
+  # ---
+  #
+  # Option +array_class+ (\Class) specifies the Ruby class to be used
+  # for each \JSON array;
+  # defaults to \Array.
+  #
+  # With the default, \Array:
+  #   source = '["foo", 1.0, true, false, null]'
+  #   ruby = JSON.parse(source)
+  #   ruby.class # => Array
+  # Use class \Set:
+  #   ruby = JSON.parse(source, {array_class: Set})
+  #   ruby # => #<Set: {"foo", 1.0, true, false, nil}>
+  # Try class \Object:
+  #   # Raises NoMethodError (undefined method `<<' for #<Object:>):
+  #   JSON.parse(source, {array_class: Object})
+  # Bad value:
+  #   # Raises TypeError (wrong argument type Symbol (expected Class)):
+  #   JSON.parse(source, {array_class: :foo})
+  #
+  # ---
+  #
+  # Option +create_additions+ (boolean) specifies whether to use \JSON additions in parsing.
+  # See {\JSON Additions}[#module-JSON-label-JSON+Additions].
+  #
+  # ====== Exceptions
+  #
+  # Raises an exception if +source+ is not \String-convertible:
+  #
+  #   # Raises TypeError (no implicit conversion of Symbol into String):
+  #   JSON.parse(:foo)
+  #
+  # Raises an exception if +opts+ is not \Hash-convertible:
+  #
+  #   # Raises TypeError (no implicit conversion of Symbol into Hash):
+  #   JSON.parse(['foo'], :foo)
+  #
+  # Raises an exception if +source+ is not valid JSON:
+  #
+  #   # Raises JSON::ParserError (783: unexpected token at ''):
+  #   JSON.parse('')
+  #
   def parse(source, opts = {})
     Parser.new(source, **(opts||{})).parse
   end
 
-  # Parse the JSON document _source_ into a Ruby data structure and return it.
-  # The bang version of the parse method defaults to the more dangerous values
-  # for the _opts_ hash, so be sure only to parse trusted _source_ documents.
-  #
-  # _opts_ can have the following keys:
-  # * *max_nesting*: The maximum depth of nesting allowed in the parsed data
-  #   structures. Enable depth checking with :max_nesting => anInteger. The
-  #   parse! methods defaults to not doing max depth checking: This can be
-  #   dangerous if someone wants to fill up your stack.
-  # * *allow_nan*: If set to true, allow NaN, Infinity, and -Infinity in
-  #   defiance of RFC 7159 to be parsed by the Parser. This option defaults
-  #   to true.
-  # * *create_additions*: If set to false, the Parser doesn't create
-  #   additions even if a matching class and create_id was found. This option
-  #   defaults to false.
+  # Calls
+  #   JSON.parse(source, opts)
```

Comment:
Technically it calls `parse`, not `JSON.parse` (e.g. if someone includes `JSON`)

Reply:
Fixed in next push.